### PR TITLE
lib: warn for clippy::use_self

### PR DESCRIPTION
**Pull request by cpu**
This was accidentally added to the `allow` lints instead of the `warn` lints in https://github.com/rustls/webpki/pull/262.

### CODE REVIEW

**Comment by cpu** [comment]
> You added it to deny instead of warn. 

:laughing: _muttering something about needing more coffee_

Fixed

File: src/lib.rs [hunk 1]
```diff
@@ -27,16 +27,15 @@
...
 #![deny(missing_docs, clippy::as_conversions)]
 #![allow(
     clippy::len_without_is_empty,
     clippy::new_without_default,
     clippy::single_match,
     clippy::single_match_else,
     clippy::type_complexity,
-    clippy::upper_case_acronyms,
-    clippy::use_self
+    clippy::upper_case_acronyms
```
Comment (ctz) [inline src/lib.rs:38]: nit: keep trailing comma?

**Review by ctz — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (ctz) [reply src/lib.rs:38]: err, assuming that's allowed by `allow()`!

Comment (cpu) [reply src/lib.rs:38]: `cargo fmt` erases it :-(

Comment (djc) [reply src/lib.rs:38]: I looked into that, unfortunately rustfmt has a bug -- there's a PR but it's languishing.

Comment (cpu) [reply src/lib.rs:38]: Bummer :-(

**Pull request merged**